packages/grafana-data/src/utils/dataFrameHelper.ts

```diff
@@ -5,16 +5,23 @@ import { ArrayVector, MutableVector, vectorToArray, CircularVector } from './vec
 import isArray from 'lodash/isArray';
 import isString from 'lodash/isString';
 
+interface FieldCacheItem {
+  field: Field;
+  idx: number;
+}
 export class FieldCache {
-  fields: Field[] = [];
+  fields: FieldCacheItem[] = [];
```

Comment:
rather than having `FieldCacheItem` with two properties... what about extending Field:
```
interface FieldCacheItem extends Field {
  index: number;
}
```

then it can just behave normally without needing the extra property reference

Reply:
Yeah, sounds good